### ISSUE TITLE
Failed to initialize SASL authentication with krb5.conf

### ISSUE THREAD

**Issue by bigkjp97** [ClickHouse/ClickHouse#39355]
Hello, i set my config.xml like👇

```xml
  <kafka>
    <security_protocol>SASL_PLAINTEXT</security_protocol>
    <sasl_kerberos_service_name>kafka</sasl_kerberos_service_name>
    <sasl_kerberos_keytab>/tmp/test.keytab</sasl_kerberos_keytab>
    <sasl_kerberos_principal>kafka/test@TEST.COM</sasl_kerberos_principal>
    <debug>cgrp</debug>
    <auto_offset_reset>smallest</auto_offset_reset>
  </kafka>
```
but i still get some errors like👇:
```text
Failed to initialize SASL authentication: SASL handshake failed (start (-1)): SASL(-1): generic failure: GSSAPI Error: No credentials were supplied, or the credentials were unavailable or inaccessible (No Kerberos credentials available: Unsupported credentials cache format version number (filename: /tmp/krb5cc_501)) (after 0ms in state AUTH_REQ, 2 identical error(s) suppressed)
```

I have set variables of 'krb5.conf' in /etc/profile, but clickhouse doesnt knw where my 'krb5.conf' is, wht should i add in 'config.xml'? And btw, how to launch a perfect clickhouse-client (with some params) in order to sync data with a kafka engine:=)

**Comment by filimonov** [comment]
Hm, maybe you can pass that via flags in sasl_kerberos_kinit_cmd? 

See https://docs.confluent.io/3.3.1/clients/librdkafka/CONFIGURATION_8md.html

/cc @ilejn

**Comment by filimonov** [comment]
You should pass it via the environment variables. 

See https://github.com/ClickHouse/ClickHouse/issues/35647#issuecomment-1188595819

**Comment by bigkjp97** [comment]
> Hm, maybe you can pass that via flags in sasl_kerberos_kinit_cmd?
> 
> See https://docs.confluent.io/3.3.1/clients/librdkafka/CONFIGURATION_8md.html
> 
> /cc @ilejn

It works, but when i create a table with kafka engine, another error appears: 
```
2022.07.19 12:00:21.458285 [ 168225 ] {09b25525-442f-4563-9c5c-69ca49fc503c} <Error> StorageKafka (test_queue): std::exception. Code: 1001, type: cppkafka::ConfigException, e.what() = Failed to set krb5.conf: No such configuration property: "krb5.conf", Stack trace (when copying this message, always include the lines below):
```

**Comment by bigkjp97** [comment]
> > Hm, maybe you can pass that via flags in sasl_kerberos_kinit_cmd?
> > See https://docs.confluent.io/3.3.1/clients/librdkafka/CONFIGURATION_8md.html
> > /cc @ilejn
> 
> It works, but when i create a table with kafka engine,
> 
> ```sql
> CREATE TABLE IF NOT EXISTS testdb.test_queue(
>                     messsage  String 
>                 ) ENGINE = Kafka
>                 SETTINGS
>                   kafka_broker_list = 'kafka-1.test.com:9092,kafka-2.test.com:9092,kafka-3.test.com:9092',
>                   kafka_topic_list = 'test_topic',
>                   kafka_group_name = 'test_cg_1',
>                   kafka_format = 'JSONAsString',
>                   kafka_row_delimiter = '\n',
>                   kafka_schema = '',
>                   kafka_num_consumers = 1
> ```
> 
> another error appears:
> 
> ```
> 2022.07.19 12:00:21.458285 [ 168225 ] {09b25525-442f-4563-9c5c-69ca49fc503c} <Error> StorageKafka (test_queue): std::exception. Code: 1001, type: cppkafka::ConfigException, e.what() = Failed to set krb5.conf: No such configuration property: "krb5.conf", Stack trace (when copying this message, always include the lines below):
> ```

nope, seems nothing work🙃

**Comment by filimonov** [comment]
> i wonder if there r some examples for kerberos kafka to clickhouse?🙃

https://github.com/ClickHouse/ClickHouse/tree/master/tests/integration/test_storage_kerberized_kafka

**Comment by ilejn** [comment]
"Unsupported credentials cache format version number" looks worrying.
Is klist working Ok ?
What is special in your krb5.conf ?

**Comment by bigkjp97** [comment]
> "Unsupported credentials cache format version number" looks worrying. Is klist working Ok ? What is special in your krb5.conf ?

Questions above all solved by using version 22.7, thanks for the replys
